tests/gui/test_widgets.py

```diff
@@ -617,6 +565,136 @@ def test_SourceWidget_delete_source_when_user_chooses_cancel(mocker, session, so
     sw.controller.delete_source.assert_not_called()
 
 
+def test_StarToggleButton_init_source_starred(mocker):
+    source = mocker.MagicMock()
+    source.is_starred = True
+
+    stb = StarToggleButton(source)
+
+    assert stb.source == source
+    assert stb.isChecked() is True
+
+
+def test_StarToggleButton_init_source_unstarred(mocker):
+    source = mocker.MagicMock()
+    source.is_starred = False
+    source.setChecked = mocker.MagicMock()
```

Comment:
Isn't `source` a db object in this test? isn't this line assuming source is a widget (since it has a `setChecked` method)? lmk if I misunderstand

Reply:
Nit: if the above is right we _could_ use the Source factory here (`factory.Source()` such that we need one less magic mock)

(not necessary for merge)

Reply:
oops, you're right, `source` is meant to be a db object and `setChecked` shouldn't be mocked. I updated this to use the Source factory.